src/main/java/lotto/domain/Winning.java

```diff
@@ -0,0 +1,56 @@
+package lotto.domain;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public enum Winning {
+    FIRST_PLACE(6, 2000000000),
+    SECOND_PLACE(5, 1500000),
+    THIRD_PLACE(4, 50000),
+    FOURTH_PLACE(3, 5000),
+    LOSE(0, 0);
+
+    private static final int MIN_WINNING_CONDITION = 3;
+
+    private static Map<Integer, Winning> winningTypes;
+
+    private int matchCount;
+    private int winningPrice;
+
+    static {
+        winningTypes = new LinkedHashMap<>();
+
+        for (Winning winning : Winning.values()) {
+            winningTypes.put(winning.getMatchCount(), winning);
+        }
+    }
+
+    public static List<Winning> winValues() {
+        return Arrays.stream(values()).filter(winning -> winning != LOSE).collect(Collectors.toList());
+    }
+
+    public static Winning getWinningType(int matchCount) {
+        if (matchCount < MIN_WINNING_CONDITION) {
+            return LOSE;
+        }
+
+        return winningTypes.get(matchCount);
+    }
+
+    Winning(int matchCount, int winningPrice) {
+        this.matchCount = matchCount;
+        this.winningPrice = winningPrice;
+    }
+
+    public int getMatchCount() {
+        return matchCount;
+    }
+
+    public int getWinningPrice() {
+        return winningPrice;
+    }
+
+    public int calculatePrice(int matchCount) {
```

Comment:
method 순서에 대해서도 한번 고민해보셔요.

https://stackoverflow.com/questions/4668218/are-there-any-java-method-ordering-conventions

Reply:
1. Class (static) variables: First the public class variables, then the protected, and then the private.
2. Instance variables: First public, then protected, and then private.
3. Constructors
4. Methods: These methods should be grouped by functionality rather than by scope or accessibility. For example, a private class method can be in between two public instance methods. The goal is to make reading and understanding the code easier.